crates/rpc/rpc/src/eth/filter.rs

```diff
@@ -38,14 +38,15 @@ use tracing::trace;
 const MAX_HEADERS_RANGE: u64 = 1_000; // with ~530bytes per header this is ~500kb
 
 /// `Eth` filter RPC implementation.
-pub struct EthFilter<Provider, Pool> {
-    /// All nested fields bundled together.
-    inner: Arc<EthFilterInner<Provider, Pool>>,
+pub struct EthFilter<Provider, Pool, D: PoolTransaction + Clone> {
```

Comment:
this is not needed, because the transaction type is an associated type of the `Pool` type

so you can use `Pool::Transaction` or `< Pool as TransactionPool>::Transaction`

Reply:
> this is not needed, because the transaction type is an associated type of the `Pool` /TransactionPool trait
> 
> so you can use `Pool::Transaction` or `< Pool as TransactionPool>::Transaction` internally

okay followed what you told, found myself modyfing lot of stuff include lib.rs eth.rs auth. startup.rs http.rs and many others files

is this normal or i'm doing something completly wrong ?

Reply:
you're right this is a bit messy, so I solved this differently

Reply:
> ou're right this is a bit messy, so I solved this differently

oh i'm really sorry for this Matt, i was thinking coming back for this another time :(

Sorry for let you doing this at my place. I was going to take some break with this and come back to it..

Sorry again